crates/nu-protocol/src/completion.rs

```diff
@@ -55,3 +56,20 @@ pub enum SuggestionKind {
     Operator,
     Variable,
 }
+
+/// A simple wrapper for [`ast::Call`] which contains additional context about completion.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct DynamicCompletionCall {
```

Comment:
I was going to suggest including `arg_type: ArgType` here, but I guess it has to be passed to plugins too. Very sad